src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -15,6 +17,9 @@ namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
     /// </summary>
     public static class TagHelperDescriptorFactory
     {
+        public static readonly ISet<char> InvalidNonWhitespaceNameCharacters = new HashSet<char>(
+            new[] { '@', '!', '<', '/', '?', '[', '>', ']', '=', '"', '\'' });
```

Comment:
1. `ISet` is mutable and that's not what we want here.  use `ICollection<char>` since `Contains()` is the only supported operation.
2. `public static readonly` isn't commonly used for collection values.  should be a `static` get-only property.